homeassistant/components/hvv_departures/binary_sensor.py

```diff
@@ -0,0 +1,189 @@
+        """Fetch data from API endpoint.
+
+        This is the place to pre-process the data to lookup tables
+        so entities can quickly look up their data.
+        """
+
+        payload = {"station": station}
+
+        try:
+            async with async_timeout.timeout(10):
+                return get_elevator_entities_from_station_information(
+                    station_name, await hub.gti.stationInformation(payload)
+                )
+        except InvalidAuth as err:
+            raise UpdateFailed(f"Authentication failed: {err}")
+        except ClientConnectorError as err:
+            raise UpdateFailed(f"Network not available: {err}")
+        except Exception as err:  # pylint: disable=broad-except
+            raise UpdateFailed(f"Error occurred while fetching data: {err}")
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="hvv_departures.binary_sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(hours=1),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        HvvDepartureBinarySensor(coordinator, idx, entry)
+        for (idx, ent) in coordinator.data.items()
+    )
+
+
+class HvvDepartureBinarySensor(BinarySensorEntity):
+    """HVVDepartureBinarySensor class."""
+
+    def __init__(self, coordinator, idx, config_entry):
+        """Initialize."""
+        self.coordinator = coordinator
+        self.idx = idx
+        self.config_entry = config_entry
+
+    @property
+    def is_on(self):
+        """Return entity state."""
+        return self.coordinator.data[self.idx]["state"]
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        return (
+            self.coordinator.last_update_success
+            and self.coordinator.data[self.idx]["available"]
+        )
+
+    @property
+    def device_info(self):
+        """Return the device info for this sensor."""
+        return {
+            "identifiers": {
+                (
+                    DOMAIN,
+                    self.config_entry.entry_id,
+                    self.config_entry.data[CONF_STATION]["id"],
+                    self.config_entry.data[CONF_STATION]["type"],
+                )
+            },
+            "name": f"Departures at {self.config_entry.data[CONF_STATION]['name']}",
+            "manufacturer": MANUFACTURER,
+        }
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self.coordinator.data[self.idx]["name"]
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return ICON
```

Comment:
We do not allow setting an icon when a device class is specified. So, either choose to use the icon and drop the device class, or the other way around.